common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -298,6 +315,17 @@ private ManagedBuffer getSortBasedShuffleBlockData(
     }
   }
 
+  public ManagedBuffer getDiskPersistedRddBlockData(
+      ExecutorShuffleInfo executor, int rddId, int splitIndex) {
+    File file = getFile(executor.localDirs, executor.subDirsPerLocalDir,
+      "rdd_" + rddId + "_" + splitIndex);
+    ManagedBuffer res = null;
+    if (file.exists()) {
+      res = new FileSegmentManagedBuffer(conf, file, 0, file.length());
```

Comment:
Yes, that was my initial idea. 

And both cases are tested:
- `ExternalShuffleIntegrationSuite#testFetchCorruptRddBlock` tests the illegal size, as when the file length is saved into val earlier still it can be deleted after the `exists()` check (or as `FileSigmentManagedBuffer` does not opens the file the file still can be deleted after the instance is constructed)
- `BlockManagerSuite#SPARK-25888: serving of removed file not detected by shuffle service`: where the 0 length file is tested (file removed right after `exist()` check)

I will revert the inlining but I still think the best to keep both tests.